### PR TITLE
Fix on demand default arrays + support for Node 4.5

### PR DESCRIPTION
**Pull request by chfritz**
1. Fixes an issue with arrays whose value is not specified during message creation, when using the onTheFly message definitions.

2. Counterpart to https://github.com/RethinkRobotics-opensource/ros_msg_utils/pull/6. Once the other PR is merged in ros_msg_utils, we can change the dependency back to the RethinkRobotics-opensource repo.

### CODE REVIEW

**Comment by chris-smith** [comment]
Ah! Sorry I broke the default arrays - comments for babel here will be the same as those over on RethinkRobotics-opensource/ros_msg_utils#6

**Comment by chfritz** [comment]
You didn't break anything. They were broken all along ;-)

File: package.json [hunk 1]
```diff
@@ -24,12 +26,14 @@
     "mocha": "^2.4.5"
   },
   "dependencies": {
+    "babel-cli": "^6.18.0",
```
Comment (chris-smith) [inline package.json]: should these be `devDependencies`?

**Comment by chfritz** [comment]
fixed. thanks.

File: package.json [hunk 2]
```diff
@@ -21,15 +23,17 @@
...
+    "babel-preset-es2015": "^6.18.0"
   },
   "dependencies": {
     "argparse": "1.0.7",
     "async": "2.0.1",
     "bunyan": "1.8.1",
     "md5": "2.1.0",
     "moment": "2.12.0",
-    "ros_msg_utils": "RethinkRobotics-opensource/ros_msg_utils#v1.0.1",
+    "ros_msg_utils": "chfritz/ros_msg_utils#add_babel",
```
Comment (chris-smith) [inline package.json]: `ros_msg_utils` has been published so you should be able to set this to 1.0.2 - I think that's the only thing left to update

**Comment by chfritz** [comment]
great! done.

**Pull request merged**